sdk/core/Azure.Core/src/Shared/ProtocolOperation.cs

```diff
@@ -30,6 +30,17 @@ internal ProtocolOperation(ClientDiagnostics clientDiagnostics, HttpPipeline pip
         public override string Id => throw new NotSupportedException();
 #pragma warning restore CA1822
 
+        /// <inheritdoc />
+        public override RehydrationToken? GetRehydrationToken()
```

Comment:
I wonder if this should be added to the OperationState so that all operation types pick up the new changes.

Reply:
I have added the method to `IOperation` and `IOperation<T>` as suggested so every operation can make use of it. Moving it to `OperationState` is a bit tricky because we'd need to pass around a couple of variables that are computed in `NextLinkOperationImplementation` (RequestMethod, _startRequestUri, _nextRequestUri, _headerSource, _lastKnownLocation, _finalStateVia, OperationId).